### PR TITLE
Update AndroidManifestFinder for Gradle builds with kapt

### PR DESCRIPTION
**Pull request by dodgex**
This should fix https://github.com/androidannotations/androidannotations/issues/2028

@WonderCsabo  @ahornerr could you test this?

### CODE REVIEW

**Comment by dodgex** [comment]
It even works for multiple flavor dimensions. :)
```
    flavorDimensions "flavor", "color"
    productFlavors {
        some_flavor {
            dimension "flavor"
        }
        some_flavor2 {
            dimension "flavor"
        }
        red {
            dimension "color"
        }
        green {
            dimension "color"
        }
    }
```

File: AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java [hunk 1]
```diff
@@ -187,10 +187,35 @@ boolean applies() {
...
 		@Override
 		Iterable<String> possibleLocations() {
+			String path = matcher.group(1);
 			String gradleVariant = matcher.group(2);
+			String variantPart = gradleVariant.substring(1);
 
-			return Arrays.asList("build/intermediates/manifests/full" + gradleVariant, "build/intermediates/bundles" + gradleVariant,
-					"build/intermediates/manifests/aapt" + gradleVariant);
+			ArrayList<String> possibleLocations = new ArrayList<>();
+			for (String directory : Arrays.asList("build/intermediates/manifests/full", "build/intermediates/bundles", "build/intermediates/manifests/aapt")) {
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java]: Add a new line.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java [hunk 2]
```diff
@@ -187,10 +187,35 @@ boolean applies() {
...
+				findPossibleLocations(path, directory, variantPart, possibleLocations);
+			}
+			return possibleLocations;
+		}
+
+		private void findPossibleLocations(String basePath, String targetPath, String variantPart, List<String> possibleLocations) {
+			String[] directories = new File(basePath + targetPath).list();
+			if (directories == null) {
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java]: Add a new line.

**Comment by dodgex** [comment]
Not sure where exaclty you want new lines.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java (hunk 1 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java]: Before the `for`

File: AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java (hunk 2 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java]: Before the `if`

**Comment by dodgex** [comment]
The new commit fixes APT builds.

**Comment by WonderCsabo** [comment]
Thanks! Can you also add tests which use flavors?

**Comment by dodgex** [comment]
I'm on it :)

**Comment by dodgex** [comment]
@WonderCsabo tests added.

**Comment by ahornerr** [comment]
What's the best way to test this?

**Comment by WonderCsabo** [comment]
@ahornerr what i did is creating flavors in the example projects.

**Comment by dodgex** [comment]
merged commits and added requested new lines.

**Comment by WonderCsabo** [comment]
I tested it again, works well, thanks!

**Pull request merged**